gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -133,45 +134,42 @@ private ObjectId convertGitCoreCommitToObjectId(IGitCoreCommit commit) throws Gi
 
   @Override
   public IGitCoreHeadSnapshot deriveHead() throws GitCoreException {
-    Ref ref;
-    try {
-      ref = jgitRepo.getRefDatabase().findRef(Constants.HEAD);
-    } catch (IOException e) {
-      throw new GitCoreException("Cannot get current branch", e);
-    }
+    Ref ref = Try.of(() -> jgitRepo.getRefDatabase().findRef(Constants.HEAD))
+        .getOrElseThrow(e -> new GitCoreException("Cannot get current branch", e));
 
     if (ref == null) {
       throw new GitCoreException("Error occurred while getting current branch ref");
     }
+
     var reflog = deriveReflogByRefFullName(Constants.HEAD);
-    IGitCoreLocalBranchSnapshot targetBranch;
 
-    String currentBranchName = null;
+    Option<String> currentBranchName = Option.none();
```

Comment:
Possibly this will be even clearer with `String currentBranchName = null;` (since you can do without any `.map` or `.flatMap` on `currentBranchName` that would merit using an `Option`, just`.isDefined()` or `.get()`

Reply:
ok